findlibs/__init__.py

```diff
@@ -146,8 +146,48 @@ def find(lib_name, pkg_name=None):
         os.path.expanduser("~/.local"),
     ):
         for lib in ("lib", "lib64"):
-            fullname = os.path.join(root, lib, libname)
+            fullname = os.path.join(root, lib, lib_name)
             if os.path.exists(fullname):
                 return fullname
+    return None
 
+def _find_in_ctypes_util(lib_name: str, pkg_name: str) -> str|None:
     return ctypes.util.find_library(lib_name)
+
+def find(lib_name: str, pkg_name: str|None = None) -> str|None:
+    """Returns the path to the selected library, or None if not found.
+
+    Arguments
+    ---------
+    lib_name : str
+        Library name without the `lib` prefix. The name of the library to
+        find is formed using ``lib_name`` and a platform specific suffix
+        (by default ".so"). E.g. when ``lib_name`` is "eccodes" the library
+        name will be "libeccodes.so" on Linux and "libeccodes.dylib"
+        on macOS.
+    pkg_name :  str, optional
+        Package name if it differs from the library name. Defaults to None.
+
+    Returns
+    --------
+    str or None
+        Path to selected library
+    """
+    pkg_name = pkg_name or lib_name
+    extension = EXTENSIONS.get(sys.platform, ".so")
+    lib_name = "lib{}{}".format(lib_name, extension)
+
+    sources = [
```

Comment:
We need a way to disable _find_in_package so that users can override it and use a system library instead. Currently, ecCodes uses an environment variable for this, which works well because it allows ECMWF's module system to take precedence. It would also be useful for developers. Perhaps something like FINDLIBS_DISABLE_FIND_IN_PACKAGE=1 ?

Reply:
right, that sounds like a neat solution. I originally thought I'd simply uninstall the package, I guess I got accustomed to `uv`'s installs/uninstalls being just hardlinking operations :)

I can even do it generically -- construct the list of `sources` by checking `FINDLIBS_DISABLE_{source}` envvar, it's not much programmatic overhead, and would make it consistent